### PR TITLE
Remove gateway-v2 & simplify engine-v2 tracing

### PR DESCRIPTION
**Pull request by Finistere**
This PR can be roughly summarized as removing the `gateway-v2` layer and a lot of the code on top. 
- Now the engine receives a `BatchRequest` and returns a `HttpGraphqlResponse` with headers and a body (bytes). The latter can be a stream if the appropriate accept header is present. This removes all the copy-pasted code between federated-dev/ federated-server and will remove a bunch in api.
- I've split the websocket implementation moving the `axum` specific parts into a `engine-v2-axum` and the rest into `engine-v2` which exposes less of the Engine internals.
- Auth is now part of the engine. By default, it just accepts everything like before. 
- I've removed the caching for now, I had a PR almost ready to replace it a few weeks ago. It was just too messy with the gateway/engine split (cache key needs to be generated after the operation parsing). Will add it back later.
- removing `plan_cache` feature, I just noticed with CI failure how I was taking correctly into account persisted queries. With stuff like `@skip` & `@include` which need to touch the operation plan, it's something I'll add later back. Not great for performance though, but well.

### CODE REVIEW

**Review by obmarg — APPROVED** [review]

File: engine/crates/engine-v2/src/http_response/mod.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        }
+        let mut body = Vec::with_capacity(
+            // '[]' + commas + actual bodies
+            2 + (bytes_batch.len() - 1) + bytes_batch.iter().map(|bytes| bytes.len()).sum::<usize>(),
+        );
+        body.push(b'[');
+        for bytes in Itertools::intersperse(bytes_batch.iter().map(|bytes| bytes.as_ref()), &[b',']) {
+            body.extend_from_slice(bytes);
+        }
+        body.push(b']');
```
[the remark this answers is not in the document]
Comment (Finistere) [reply engine/crates/engine-v2/src/http_response/mod.rs:92]: good point

File: engine/crates/engine-v2/src/engine.rs [hunk 2]
```diff
@@ -1,182 +1,279 @@
...
+            .map(move |response| match response {
+                Response::BadRequest(_) => websocket::Message::Error {
+                    id: id.clone(),
+                    payload: websocket::Payload(response),
+                },
+                response => websocket::Message::Next {
+                    id: id.clone(),
+                    payload: websocket::Payload(response),
+                },
+            })
```
[the remark this answers is not in the document]
Comment (Finistere) [reply engine/crates/engine-v2/src/engine.rs:277]: yeah good catch, thanks

File: engine/crates/engine-v2/axum/src/websocket/accepter.rs [hunk 3]
```diff
@@ -241,25 +214,18 @@ async fn accept_websocket(
...
         while let Some(message) = self.recv().await {
-            let event = match message {
+            return match message {
                 Ok(ws::Message::Ping(_) | ws::Message::Pong(_)) => continue,
-                Ok(ws::Message::Close(_)) => return None,
-                Ok(ws::Message::Text(contents)) => serde_json::from_str::<Event>(&contents),
-                Ok(ws::Message::Binary(contents)) => serde_json::from_slice::<Event>(&contents),
+                Ok(ws::Message::Close(_)) => None,
+                Ok(ws::Message::Text(contents)) => Some(contents.into_bytes()),
+                Ok(ws::Message::Binary(contents)) => Some(contents),
```
[the remark this answers is not in the document]
Comment (Finistere) [reply engine/crates/engine-v2/axum/src/websocket/accepter.rs:223]: would need to handle invalid UTF-8 errors at this stage though, might as well leave `serde_json` handle this during deserialization imho

File: Cargo.toml [hunk 4]
```diff
@@ -15,6 +16,7 @@ members = [
   "graph-ref",
   "graphql-lint",
 ]
+exclude = ["engine/crates/gateway-v2"]
```
Comment (obmarg) [inline Cargo.toml:19]: Why?

File: engine/crates/engine-v2/axum/src/websocket/accepter.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (obmarg) [reply engine/crates/engine-v2/axum/src/websocket/accepter.rs:218]: I'd even be fine with just rejecting binary messages tbh.  I only added support for them for completeness - writing the code around them at the expense of the messages that will actually come in seems bad.

File: Cargo.toml (hunk 4 above)
Comment (obmarg) [reply Cargo.toml:19]: k

File: engine/crates/engine-v2/axum/src/websocket/accepter.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Finistere) [reply engine/crates/engine-v2/axum/src/websocket/accepter.rs:218]: not sure how it's at the expense of others, I just don't see any value to force a `String` here imho

**Pull request merged**